src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -837,7 +838,7 @@ private static IntermediateNode GetBooleanBinaryOp(IRTranslatorContext context,
                     }
                     else
                     {
-                        throw new NotSupportedException();
+                        return new BooleanLiteralNode(context.GetIRContext(node), false);
```

Comment:
Add a test in IRTranslatorTests.cs that validates we produce a false literal node for comparisons that don't coerce?

Reply:
Done